### PR TITLE
add params passing and a test

### PR DESCRIPTION
**Pull request by engineerkatie**

### CODE REVIEW

File: libs/dmlf/include/dmlf/execution/execution_params.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "variant/variant.hpp"
+#include <vector>
+
+using ExecutionParams = std::vector<fetch::variant::Variant>;
```
Comment (lrahmani) [inline libs/dmlf/include/dmlf/execution/execution_params.hpp]: I would rather alias `Variant` than `vector<Variant>`, as the user will have to handle `Variants` anyway 

```cpp
using Parameter = fetch::variant::Variant;
using ExecutionParams = std::vector<Parameter>
```
user code can than do 

```cpp
uto p3 = client->client_->Run(SERVER_PUB, "exe1", "state1", "add", {Parameter{"hi"}, Parameter{2}});
```

Then we could make `Parameter` what ever we would it like to be, if that is the reason for having an alias in the first place

Comment (engineerkatie) [reply libs/dmlf/include/dmlf/execution/execution_params.hpp]: I know what you mean but my thinking was that the function we're calling takes a list of parameters, so the "top level" vector represents the actual parameter list (some of which may then themselves be vectors).

Comment (engineerkatie) [reply libs/dmlf/include/dmlf/execution/execution_params.hpp]: Oh, no, hang on, I think I've misunderstood you.

Comment (engineerkatie) [reply libs/dmlf/include/dmlf/execution/execution_params.hpp]: Yes, I have, I see what you mean. Yes.

Comment (lrahmani) [reply libs/dmlf/include/dmlf/execution/execution_params.hpp]: sorry, I was not very clear in my comment.
Yes, I meant that `Run` method doesn't need to abstract away the fact that parameters are a list/vector of elements. The elements could be named Variant, Parameter, Variable or anything else, it doesn't matter as long as we provide a practical (generic, flexible) interface to set and get values of different types out of them.

Comment (engineerkatie) [reply libs/dmlf/include/dmlf/execution/execution_params.hpp]: Done!

File: libs/dmlf/include/dmlf/remote_execution_client.hpp.bak [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#pragma once
```
Comment (lrahmani) [inline libs/dmlf/include/dmlf/remote_execution_client.hpp.bak]: note: can be deleted from this web page

File: libs/dmlf/include/dmlf/remote_execution_host.hpp.bak [hunk 3]
```diff
@@ -0,0 +1,83 @@
+#pragma once
```
Comment (lrahmani) [inline libs/dmlf/include/dmlf/remote_execution_host.hpp.bak]: note: can be deleted from this web page

File: libs/dmlf/include/dmlf/execution/execution_interface.hpp.bak [hunk 4]
```diff
@@ -0,0 +1,60 @@
+#pragma once
```
Comment (lrahmani) [inline libs/dmlf/include/dmlf/execution/execution_interface.hpp.bak]: note: can be deleted from this web page

**Review by lrahmani — COMMENTED** [review]
lgtm

File: libs/dmlf/include/dmlf/remote_execution_client.hpp.bak (hunk 2 above)
Comment (engineerkatie) [reply libs/dmlf/include/dmlf/remote_execution_client.hpp.bak]: (sigh)

Gotta add these to the gitignore...

**Review by lrahmani — APPROVED** [review]

**Pull request merged**